control/controlclient/netmap.go

```diff
@@ -39,6 +39,12 @@ type NetworkMap struct {
 	Hostinfo      tailcfg.Hostinfo
 	PacketFilter  []filter.Match
 
+	// ServicesCollectionAllowed is whether the node's Tailnet has
+	// opted into allowing collection of services information. When
+	// false, Services are excluded from Hostinfo. Hostinfo.ShieldsUp
```

Comment:
You might explain in what way ShieldsUp takes precedence. (Maybe something along the lines of: "If set, ShieldsUp blocks services collection; that takes precedence over this field.)